src/plugins/ui_actions/public/components/action_wizard/action_wizard.tsx

```diff
@@ -0,0 +1,219 @@
+        }}
+        onConfigChange={newConfig => {
+          onChange(selectedActionFactory, newConfig);
+        }}
+      />
+    );
+  }
+
+  return (
+    <ActionFactorySelector
+      actionFactories={actionFactories}
+      onActionFactorySelected={actionFactory => {
+        setSelectedActionFactory(actionFactory);
+        onChange(actionFactory, null);
+      }}
+    />
+  );
+};
+
+interface SelectedActionFactoryProps<Config extends ActionFactoryBaseConfig, Context = unknown> {
+  actionFactory: ActionFactory<Config, Context>;
+  onConfigChange: (config: Config | null) => void;
+  showDeselect: boolean;
+  onDeselect: () => void;
+}
+export const TEST_SUBJ_SELECTED_ACTION_FACTORY = 'selected-action-factory';
+const SelectedActionFactory: React.FC<SelectedActionFactoryProps<ActionFactoryBaseConfig>> = ({
+  actionFactory,
+  onDeselect,
+  showDeselect,
+  onConfigChange,
+}) => {
+  const [config, setConfig] = useState<ActionFactoryBaseConfig | null>(null);
+  return (
+    <div
+      className="uiActions__SelectedActionFactory"
+      data-test-subj={TEST_SUBJ_SELECTED_ACTION_FACTORY}
+    >
+      <header>
+        <EuiFlexGroup alignItems="center" gutterSize="s">
+          {actionFactory.iconType && (
+            <EuiFlexItem grow={false}>
+              <EuiIcon type={actionFactory.iconType} size="m" />
+            </EuiFlexItem>
+          )}
+          <EuiFlexItem grow={true}>
+            <EuiText>
+              <h4>{actionFactory.displayName}</h4>
+            </EuiText>
+          </EuiFlexItem>
+          {showDeselect && (
+            <EuiFlexItem grow={false}>
+              <EuiButtonEmpty size="s" onClick={() => onDeselect()}>
+                {txtChangeButton}
+              </EuiButtonEmpty>
+            </EuiFlexItem>
+          )}
+        </EuiFlexGroup>
+      </header>
+      <EuiSpacer size="m" />
+      <div>
+        {actionFactory.wizard({
+          context: actionFactory.context,
+          config,
+          onConfig: newConfig => {
+            setConfig(newConfig);
+            onConfigChange(newConfig);
+          },
+        })}
+      </div>
+    </div>
+  );
+};
+
+interface ActionFactorySelectorProps {
+  actionFactories: Array<ActionFactory<ActionFactoryBaseConfig, unknown>>;
+  onActionFactorySelected: (actionFactory: ActionFactory<ActionFactoryBaseConfig, unknown>) => void;
+}
+export const TEST_SUBJ_ACTION_FACTORY_ITEM = 'action-factory-item';
+const ActionFactorySelector: React.FC<ActionFactorySelectorProps> = ({
+  actionFactories,
+  onActionFactorySelected,
+}) => {
+  if (actionFactories.length === 0) {
+    // this is not user facing, as it would be impossible to get into this state
+    // just leaving for dev purposes for troubleshooting
+    return <div>No action factories to pick from</div>;
+  }
+
+  return (
+    <EuiFlexGroup wrap={true}>
+      {actionFactories.map(actionFactory => (
+        <EuiFlexItem
+          className="uiActions__ActionFactory"
```

Comment:
When creating classnames, we try to use the following pattern: 
```
{3 letter prefix}{NameOfFile/Component}__{childElement}
```

So in this case, I would start a prefix for the `ui_actions`  as `uia`, then this classname would be:
```suggestion
          className="uiaActionWizard__actionFactoryItem"
```

Reply:
thanks, fixed